### PR TITLE
Fixes #7091: The uuid in the promises and the uuid in /opt/rudder/etc/uuid.hive may be out of sync, and chaos and sadness follows

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/7091

### CODE REVIEW

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

File: techniques/system/common/1.0/update.st [hunk 1]
```diff
@@ -349,6 +349,23 @@ bundle agent update_reports
...
+{
+  classes:
+    # Abort if no uuid is defined
+    "should_not_continue" expression => "wrong_uuid|no_uuid";
+    "wrong_uuid" not => strcmp("&UUID&", "${g.uuid}");
+    "no_uuid" not => fileexists("${g.uuid_file}");
+
+  reports:
+    wrong_uuid::
+      "FATAL: The UUID in ${g.uuid_file} (${g.uuid}) doesn't match the UUID in your promises (&UUID&)";
```
Comment (jooooooon) [inline techniques/system/common/1.0/update.st]: Could you replace "in your promises" by "recorded for this node in your Rudder server" ?

File: techniques/system/common/1.0/update.st [hunk 2]
```diff
@@ -349,6 +349,23 @@ bundle agent update_reports
...
+    no_uuid::
+      "FATAL: The UUID file ${g.uuid_file} is missing";
```
Comment (jooooooon) [inline techniques/system/common/1.0/update.st]: Can you add "Please run check-rudder-agent" to generate a new one or restore it from a backup." ?

**Comment by peckpeck** [comment]
Commit modified

File: initial-promises/node-server/promises.cf [hunk 3]
```diff
@@ -70,7 +70,7 @@ bundle common va
 
   vars:
     policy_server::
-      "bs" slist => { "startExecution", "check_disable_agent", "clean_red_button", "update", "set_red_button", "internal_security", "process_matching", "check_cf_processes", "check_uuid", "check_cron_daemon", "garbage_collection", "check_binaries_freshness"
+      "bs" slist => { "startExecution", "check_disable_agent", "clean_red_button", "update", "set_red_button", "internal_security", "process_matching", "check_cf_processes", "check_cron_daemon", "garbage_collection", "check_binaries_freshness"
```
Comment (jooooooon) [inline initial-promises/node-server/promises.cf]: We should keep the call to check_uuid and port the uuid checking code from techniques

**Comment by peckpeck** [comment]
Commit modified

**Comment by jooooooon** [comment]
Looks good to me. But let's wait until the current releases are OK before merging.

File: techniques/system/common/1.0/failsafe.st [hunk 4]
```diff
@@ -93,8 +95,6 @@ bundle common g
 bundle common rudder_roles
 {
   classes:
-      # Abort if no uuid is defined
-      "should_not_continue" not        => fileexists("${g.uuid_file}");
```
Comment (jooooooon) [inline techniques/system/common/1.0/failsafe.st]: This shuold also be removed in initial promises

File: techniques/system/common/1.0/promises.st [hunk 5]
```diff
@@ -168,9 +168,6 @@ bundle common rudder_roles
 {
 
   classes:
-      # Abort if no uuid is defined
-      "should_not_continue" not => fileexists("${g.uuid_file}");
```
Comment (jooooooon) [inline techniques/system/common/1.0/promises.st]: This shuold also be removed in initial promises
 Add a line note

**Comment by peckpeck** [comment]
Commit modified

**Pull request merged**